File: problem15/p15.c
#include <stdio.h>
#include <stdlib.h>

 float lattice_paths(int width, int height){
    int i, j = width;
    float result = 1;
    for(i=width+height; i>(width+height)-width; i--){
        result *= i;
        result /= j;
        j--;
    }
    while(j > 1){
        result /= j;
        j--;
    }
    return result;
}

int main()
{
    printf("%lld", (long long)lattice_paths(20, 20));
    return 0;
}

File: problem15/p15.java
public class p15 {
	
	public static double lattice_paths(int width, int height) {
		
		int i, j = width;
		double result = 1;
		for(i=width+height; i>(width+height)-width; i--) {
			result *= i;
			result /= j;
			j--;
		}
		while(j > 1) {
			result /= j;
			j--;
		}
		
		return result;
		
	}

	public static void main(String[] args) {
		
		System.out.println(lattice_paths(20, 20));
		
	}
	
}
